addons/uom/models/uom_uom.py

```diff
@@ -54,6 +66,27 @@ def _onchange_uom_type(self):
         if self.uom_type == 'reference':
             self.factor = 1
 
+    @api.constrains('category_id', 'uom_type', 'active')
+    def _check_reference_per_category_existance(self):
+        """ Force the existence of only one UoM reference per category
+            NOTE: this is a constraint on the all table. This might not be a good practice, but this is
+            not possible to do it in SQL directly.
+        """
+        category_ids = self.mapped('category_id').ids
+        self._cr.execute("""
+            SELECT C.id AS category_id, count(U.id) AS uom_count
+            FROM uom_category C
```

Comment:
Why C and not category ?

Reply:
Purpose of alias in SQL is to write less. If the gain is only 4 letters with the alias, I'd rather not used it. Also, in my database bible (https://staff.info.unamur.be/dbm/Documents/Ouvrages/Dunod-2015/BD-2015-PremieresPages.pdf), many aliases with an uppercase are used. Old habits die hard ;)

Reply:
Are you paid more by stripping characters ?

Reply:
Not sure anymore about the contract I signed some years ago. The name was "OpenERP", but it is floue ....